config.tf

```
@@ -8,7 +8,7 @@ data "azurerm_subnet" "app_gw" {
 resource "azurerm_public_ip" "app_gw" {
   count = length(var.frontends) != 0 ? 1 : 0
 
-  name                = "aks-fe-appgw-${var.env}-pip"
+  name                = (var.resource_prefix != null) ? "${var.resource_prefix}-aks-fe-appgw-${var.env}-pip" : "aks-fe-appgw-${var.env}-pip"
```

i would do:

```
locals {
 resource_prefix = var.resource_prefix != null ? "${var.resource_prefix}-" : ""
}
```

then 

```suggestion
  name                = "${var.resource_prefix}aks-fe-appgw-${var.env}-pip"
```